actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -233,7 +233,7 @@ def process(method, path, params: nil, headers: nil, env: nil, xhr: false, as: n
 
         request_env = {
           :method => method,
-          :params => request_encoder.encode_params(params),
+          :params => params.nil? ? nil : request_encoder.encode_params(params),
```

Comment:
I would remove it from here since this is a specific usage of the encode_params which already handles it.

Reply:
as i mentioned in tsolar reply that was the case, but i forgot to commit.